app/lib/actions/katello/content_view/publish.rb

```diff
@@ -9,11 +9,26 @@ class Publish < Actions::EntryAction
         def plan(content_view, description = "", options = {})
           action_subject(content_view)
           content_view.check_ready_to_publish!
+
+          if options[:repos_units].present?
+            valid_labels_from_cv = content_view.library_repos.map { |r| r.label }
+            labels_from_repos_units = options[:repos_units].map { |l| l[:repo_label] }
+
+            labels_from_repos_units.each do |l|
+              fail _("Repository label '%s' is not associated with content view.") % l unless valid_labels_from_cv.include? l
+            end
+
+            valid_labels_from_cv.each do |l|
+              fail _("Content view has repository label '%s' which is not specified in repos_units parameter.") % l unless labels_from_repos_units.include? l
+            end
+          end
```

Comment:
Can we please avoid one letter variable names. Even within a simple map or each statement, the readability is decreased with their use.